File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/projection/enrollmentlist/EnrollmentListItem.java
package cloud.ambar.creditcard.enrollment.projection.enrollmentlist;

import lombok.*;

import java.time.Instant;

@Builder
@Getter
public class EnrollmentListItem {
    @NonNull private String id;
    @NonNull private String userId;
    @NonNull private String productId;
    @NonNull private String productName;
    @NonNull private Instant requestedDate;
    @NonNull private String status;
    private String statusReason;
    private Instant reviewedOn;
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCard/Enrollment/Projection/EnrollmentList/EnrollmentListItem.cs
namespace CreditCardEnrollment.CreditCard.Enrollment.Projection.EnrollmentList;

public class EnrollmentListItem {
    public required string Id { get; init; }
    public required string UserId { get; init; }
    public required string ProductId { get; init; }
    public required string ProductName { get; init; }
    public required DateTime RequestedDate { get; init; }
    public required string Status { get; init; }
    public string? StatusReason { get; init; }
    public DateTime? ReviewedOn { get; init; }
}